headers.c

```diff
@@ -194,10 +200,13 @@ GC_INNER void GC_init_headers(void)
 {
     unsigned i;
 
-    GC_all_nils = (bottom_index *)GC_scratch_alloc(sizeof(bottom_index));
-    if (GC_all_nils == NULL) {
-      GC_err_printf("Insufficient memory for GC_all_nils\n");
-      EXIT();
+    if (GC_all_nils == NULL)
```

Comment:
This change in GC_init_headers is useless because GC_all_nils is a part of GC_arrays, thus it is guaranteed to be zero after GC_deinit.

Reply:
It has been a while since I made this change, but I recall seeing it be non-null in some cases.